client/my-sites/plans-grid/index.tsx

```diff
@@ -24,9 +26,19 @@ import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
 import type { IAppState } from 'calypso/state/types';
 import './style.scss';
 
+/*
+ *	TODO clk: On PlansGridProps:
+ *	1.	These props need to be split into what's needed separately for the internal grids and what's needed for the exported wrappers.
+ *		This is currently also used as the internal type for the FeaturesGrid (wrongly).
+ *			- onUpgradeClick is only needed for the wrappers exported from here. Internally the grids take a transformed
+ *			handleUpgradeClick/onUpgradeClick function (force renamed to handleUpgradeClick in FeaturesGrid)
+ *			- allFeaturesList is only relevant for the ComparisonGrid
+ *			- gridPlanForSpotlight is only relevant for the FeaturesGrid
+ *			- etc.
+ *	2.	Explicitly set optional props and default values. Only absolutely vital props should be required.
+ */
```

Comment:
This we ought to address in a bit of immediate follow-up. We may consider splitting these props across several pieces, and updating the respective READMEs while at it, possibly:

1. `ContextProps` (not exported) all the props that pertain to things added to the context
2. `SharedPlansGridProps` (not exported) all the props that are shared between the FeaturesGrid and ComparisonGrid (et al)
3. `ComparisonGridProps`
4. `FeaturesGridProps`

Reply:
Created a separate issue for this https://github.com/Automattic/wp-calypso/issues/82335